src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable react/require-default-props */
+/* eslint-disable no-unused-vars */
+import React, { MouseEvent, ReactNode } from 'react';
+import { Dialog, Transition } from '@headlessui/react';
+
+import ModalHeader from './components/ModalHeader';
+
+interface ModalProps {
+  open: boolean;
+  onClose: (event?: MouseEvent<HTMLElement>) => void;
+  title?: string;
+  headerAction?: ReactNode;
+  children?: ReactNode;
+  className?: string;
+}
+
+const Modal = ({
+  open,
+  onClose,
+  title,
+  headerAction,
+  children,
```

Comment:
Is the footer going to be in the children? Maybe we could create an example to show how it works with the footer.